public/static/js/legacyMain.js

```diff
@@ -80,14 +80,40 @@ const listenScrollToTopButton = () => {
   });
 };
 
-const detectEnviromentAndSetDownloadOptions = () => {
+const detectEnviromentAndSetDownloadOptions = async () => {
+  /**
+   * This function is used to get the CPU bitness of the user's device
+   * on Windows 11. For more information, see:
+   *
+   * @see
+   * [MDN](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData/getHighEntropyValues)
+   * and
+   * [MSFT](https://learn.microsoft.com/en-us/microsoft-edge/web-platform/how-to-detect-win11)
+   *
+   * @returns
+   * 32 or 64, depending on the CPU bitness of the user's device.
+   * If not supported, it returns undefined.
+   */
+  async function getCPUBitnessForWin11() {
+    if (
+      navigator.userAgentData &&
+      navigator.userAgentData.getHighEntropyValues
+    ) {
+      return await navigator.userAgentData.getHighEntropyValues(['bitness']);
+    }
+    return undefined;
+  }
+
+  const getWin11CPUProcessorBitness = getCPUBitnessForWin11();
+
   const userAgent = navigator.userAgent;
   const osMatch = userAgent.match(/(Win|Mac|Linux)/);
-
   const os = (osMatch && osMatch[1]) || '';
 
   const arch =
-    userAgent.match(/x86_64|Win64|WOW64/) || navigator.cpuClass === 'x64'
+    userAgent.match(/x86_64|Win64|WOW64/) ||
+    userAgent.cpuClass === 'x64' ||
```

Comment:
There is no such thing as `userAgent.cpuClass`, in the past there existed a `navigator.oscpu` and `navigator.cpuClass` but they were long time removed from all modern browsers. IE6-11 might actually still use it. But then keep it as `navigator.cpuClass`, please, and add a note that this is an IE11 specific attribute.